index.js

```diff
@@ -15,9 +15,23 @@ module.exports = async function serve (drive, opts = {}) {
       return
     }
 
-    const { pathname } = new URL(req.url, 'http://localhost')
+    const { pathname, searchParams } = new URL(req.url, 'http://localhost')
+    const version = searchParams.get('checkout')
+    const snapshot = version ? drive.checkout(version) : drive
     const filename = decodeURI(pathname)
-    const entry = await drive.entry(filename)
+
+    let entry
+    try {
+      entry = await snapshot.entry(filename)
+    } catch (e) {
+      const msg = e.code || e.message
+
+      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') res.writeHead(404, msg)
+      else res.writeHead(500, msg)
+
+      res.end()
```

Comment:
I think my code has a bug, the response data is empty (it doesn't contain e.code or message)
`writeHead` should not use `msg`, it should be `res.end(msg)`